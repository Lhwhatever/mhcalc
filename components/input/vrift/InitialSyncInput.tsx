import { MenuItem, TextField } from '@material-ui/core'
import React from 'react'
import { InputEvent } from '../types'

export type InitialSync = 40 | 50 | 60 | 70 | 80 | 90 | 100

export interface SyncInputProps {
    fullWidth?: boolean
    value?: InitialSync
    onChange?: (event: InputEvent) => void
}

const InitialSyncInput = (props: SyncInputProps): JSX.Element => {
    return (
        <TextField select label="Sync" variant="outlined" defaultValue={100} {...props}>
            {Array(7)
                .fill(0)
                .map((_, i) => {
                    const sync = i * 10 + 40
                    return (
                        <MenuItem key={i} value={sync}>
                            Lvl. {i + 1} ({sync} Sync)
                        </MenuItem>
                    )
                })}
        </TextField>
    )
}

export default InitialSyncInput
