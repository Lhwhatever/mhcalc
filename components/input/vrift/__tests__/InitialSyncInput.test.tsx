import { MenuItem, TextField } from '@material-ui/core'
import { createShallow } from '@material-ui/core/test-utils'
import { ShallowWrapper } from 'enzyme'
import React from 'react'
import InitialSyncInput from '../InitialSyncInput'

describe('SyncInput test', () => {
    let shallow
    let wrapper

    beforeAll(() => {
        shallow = createShallow()
    })

    beforeEach(() => {
        wrapper = shallow(<InitialSyncInput />)
    })

    it('should use TextField[select][label=Sync]', () => {
        expect(wrapper.find(TextField)).toHaveLength(1)
        expect(wrapper.find(TextField).filter('[select][label="Sync"]')).toHaveLength(1)
    })

    it('should have options for Lvl. 1-7 (40-100 Sync)', () => {
        expect(wrapper.find(MenuItem)).toHaveLength(7)
        wrapper.find(MenuItem).forEach((option: ShallowWrapper, i: number) => {
            const sync = i * 10 + 40
            expect(option.prop('value')).toStrictEqual(sync)
            expect(option.text()).toMatch(`Lvl. ${i + 1}`)
            expect(option.text()).toMatch(`${sync} Sync`)
        })
    })

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
