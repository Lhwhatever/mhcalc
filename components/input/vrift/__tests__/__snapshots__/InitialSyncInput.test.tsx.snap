// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncInput test should match snapshot 1`] = `
<WithStyles(ForwardRef(TextField))
  defaultValue={100}
  label="Sync"
  select={true}
  variant="outlined"
>
  <WithStyles(ForwardRef(MenuItem))
    key="0"
    value={40}
  >
    Lvl. 
    1
     (
    40
     Sync)
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="1"
    value={50}
  >
    Lvl. 
    2
     (
    50
     Sync)
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="2"
    value={60}
  >
    Lvl. 
    3
     (
    60
     Sync)
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="3"
    value={70}
  >
    Lvl. 
    4
     (
    70
     Sync)
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="4"
    value={80}
  >
    Lvl. 
    5
     (
    80
     Sync)
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="5"
    value={90}
  >
    Lvl. 
    6
     (
    90
     Sync)
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="6"
    value={100}
  >
    Lvl. 
    7
     (
    100
     Sync)
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(TextField))>
`;
